File: backend/ride-share/src/main/java/mk/ukim/finki/rideshare/web/request/CreateRideRequest.java
package mk.ukim.finki.rideshare.web.request;

import java.time.LocalDateTime;

public record CreateRideRequest(
        String origin,
        Double originLatitude,
        Double originLongitude,
        String destination,
        Double destinationLatitude,
        Double destinationLongitude,
        Boolean isDoorToDoor,
        LocalDateTime departureTime,
        Boolean isDepartureTimeFlexible,
        Double price,
        Boolean hasLuggageSpace,
        Integer capacity,
        Boolean isInstantBookingEnabled
) {
}

File: frontend/ride-share/src/interfaces/request/CreateRideRequest.ts
export interface CreateRideRequest {
    origin: string;
    originLatitude: number;
    originLongitude: number;
    destination: string;
    destinationLatitude: number;
    destinationLongitude: number;
    isDoorToDoor: boolean;
    departureTime?: string;
    isDepartureTimeFlexible: boolean;
    price: number;
    hasLuggageSpace: boolean;
    capacity: number;
    isInstantBookingEnabled: boolean;
}